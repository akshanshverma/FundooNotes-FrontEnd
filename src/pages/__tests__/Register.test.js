import React from 'react';
import { shallow } from 'enzyme';

import Register from "../Register";

const component = shallow(<Register debug />);
describe('Register', () => {
    it('should render correctly in "debug" mode', () => {      
        expect(component).toMatchSnapshot();
    });

    // it('renders a email input', () => {
    //     expect(component.find('#rUsername').length).toEqual(1);
    // })

    // it('renders a password input', () => {
    //     expect(component.find('#rEmail').length).toEqual(1);
    // })
    // it('renders a password input', () => {
    //     expect(component.find('#rPassword').length).toEqual(1);
    // })
    // it('renders a check password input', () => {
    //     expect(component.find('#rRPassword').length).toEqual(1);
    // })

    it('should change event and password set in state ', () => {
        const emailInput = component.find('#rUsername');
        emailInput.props().onChange({
            target: {
                name: 'username',
                value: 'akshansh'
            }
        });
        expect(component.state('username')).toEqual('akshansh');
    });
    it('should change event and password set in state ', () => {
        const emailInput = component.find('#rEmail');
        emailInput.props().onChange({
            target: {
                name: 'email',
                value: 'akshansh.verma@gmail.com'
            }
        });
        expect(component.state('email')).toEqual('akshansh.verma@gmail.com');
    });
    it('should change event and password set in state ', () => {
        const emailInput = component.find('#rPassword');
        emailInput.props().onChange({
            target: {
                name: 'password',
                value: 'qwerty'
            }
        });
        expect(component.state('password')).toEqual('qwerty');
    });
    it('should change event and password set in state ', () => {
        const emailInput = component.find('#rRPassword');
        emailInput.props().onChange({
            target: {
                name: 'rpassword',
                value: 'qwerty'
            }
        });
        expect(component.state('rpassword')).toEqual('qwerty');
    });
});