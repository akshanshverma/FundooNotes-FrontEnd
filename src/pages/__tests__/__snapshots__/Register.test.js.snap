// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register should render correctly in "debug" mode 1`] = `
<div
  className="maindiv"
>
  <WithStyles(Card)
    id="card"
  >
    <WithStyles(Typography)
      color="primary"
      id="registerT"
    >
      Register
    </WithStyles(Typography)>
    <div
      className="hold"
    >
      <div
        className="input"
      >
        <Input
          label="username"
          name="username"
          onChange={[Function]}
          placeholder="username"
          type="text"
        />
        <div
          className="msg"
        />
      </div>
      <div
        className="input"
      >
        <Input
          label="email"
          name="email"
          onChange={[Function]}
          placeholder="enter email"
          type="text"
        />
        <div
          className="msg"
        />
      </div>
      <div
        className="input"
      >
        <Input
          label="password"
          name="password"
          onChange={[Function]}
          placeholder="enter password"
          type="password"
        />
        <div
          className="msg"
        />
      </div>
      <div
        className="input"
      >
        <Input
          label="confirm password"
          name="rpassword"
          onChange={[Function]}
          placeholder="confirm password"
          type="password"
        />
        <div
          className="msg"
        />
      </div>
      <div
        className="button"
      >
        <WithStyles(Button)
          color="primary"
          onClick={[Function]}
          variant="contained"
        >
          Submit
        </WithStyles(Button)>
      </div>
    </div>
  </WithStyles(Card)>
</div>
`;
